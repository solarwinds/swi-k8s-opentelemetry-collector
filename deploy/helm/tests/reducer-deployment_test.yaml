# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for reducer-deployment
templates:
  - network/reducer-deployment.yaml
  - network/configmap.yaml
tests:
  - it: Reducer spec should match snapshot when using default values
    template: network/reducer-deployment.yaml
    set:
      ebpfNetworkMonitoring.enabled: true
    asserts:
      - matchSnapshot:
          path: spec.template.spec