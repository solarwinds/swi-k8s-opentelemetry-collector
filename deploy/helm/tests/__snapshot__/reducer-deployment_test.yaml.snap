Reducer spec should match snapshot when using default values:
  1: |
    containers:
      - args:
          - --port=7000
          - --log-console
          - --no-log-file
          - --warning
          - --enable-aws-enrichment
          - --enable-id-id
          - --disable-prometheus-metrics
          - --enable-otlp-grpc-metrics
          - --otlp-grpc-metrics-host=RELEASE-NAME-swo-k8s-collector-collector
          - --otlp-grpc-metrics-port=4317
          - --num-ingest-shards=1
          - --num-matching-shards=1
          - --num-aggregation-shards=1
        env:
          - name: EBPF_NET_CRASH_METRIC_HOST
            value: RELEASE-NAME-swo-k8s-collector-collector
          - name: EBPF_NET_CRASH_METRIC_PORT
            value: "4317"
        image: otel/opentelemetry-ebpf-reducer:v0.10.0
        imagePullPolicy: Always
        name: reducer
        ports:
          - containerPort: 7000
            name: telemetry
            protocol: TCP
        readinessProbe:
          exec:
            command:
              - /srv/health_check.sh
              - readiness_probe
              - localhost
              - "7000"
          initialDelaySeconds: 5
          periodSeconds: 20
          timeoutSeconds: 5
        resources: null
        volumeMounts: null
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: linux
    volumes: null
